### ISSUE TITLE
🤖 [AUTO] Implement Strategy: neural_network_spy_daily_returns_simplest_robust

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#451]
## 🤖 Auto-Generated Strategy Request

**Source:** Artificial Intelligence Techniques 2017.pdf (strategy 8)
**Extracted:** 2025-12-25T02:46:15.129531

---

1.  **Strategy Name**: neural_network_spy_daily_returns_simplest_robust
2.  **Strategy Type**: Regression, Directional
3.  **Timeframe**: 1D (Holding for one day, entering at close, exiting at next close)
4.  **Instruments**: SPY (S&P 500 ETF)

5.  **Entry Rules**:
    *   **Long Entry**: Enter a long position if the averaged predicted future one-day return from the neural network model is positive.
    *   **Short Entry**: Enter a short position if the averaged predicted future one-day return from the neural network model is negative.

6.  **Exit Rules**:
    *   **Take Profit**: Not explicitly defined, but implied by 1-day holding period.
    *   **Stop Loss**: Not explicitly defined.
    *   **Time-based Exit**: All positions are held for one trading day.

7.  **Indicators Required**:
    *   Previous 1-day return (`ret1`)
    *   Previous 2-day return (`ret2`)
    *   Previous 5-day return (`ret5`)
    *   Previous 20-day return (`ret20`)
    *   Feed-forward Neural Network (MATLAB `feedforwardnet` function).
    *   Architecture: One hidden layer with one neuron (`hiddenSizes=[1]`). This configuration was identified as the most robust to avoid overfitting for this problem.
    *   Training: Averaging the predicted returns of multiple (e.g., 100) networks, each trained with different initial guesses for weights and different data splits (60% train, 40% validation for early stopping, 0% internal test set).

8.  **Risk Management**:
    *   **Position Sizing**: Fixed capital amount per trade.
    *   **Maximum Risk per Trade**: Not specified.
    *   **Risk-Reward Ratio**: Not specified.

9.  **Additional Notes**:
    *   The text emphasizes the high risk of overfitting with neural networks, especially with limited data and predictors.
    *   More complex architectures (more layers, more neurons) were shown to increase in-sample performance at the expense of severe out-of-sample degradation.
    *   The training data was from Dec 22, 2004, to Sep 15, 2009. The test data was from Sep 16, 2009, to Jun 2, 2014.
    *   Performance (1 neuron, 1 hidden layer, averaging): 26% CAGR (in-sample) and 1.9% CAGR (out-of-sample) with a -0.57 Sharpe ratio. This indicates very weak, likely non-significant out-of-sample performance.
    *   The conclusion is that, for this specific problem with limited features, neural networks do not perform well, reinforcing the idea that "complexity doesn't pay" in trading.

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/neural_network_spy_daily_returns_simplest_robust.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
